### ISSUE TITLE
`QUnit.module.only` behaves strange

### ISSUE THREAD

**Issue by sechel** [qunitjs/qunit#1272]
`QUnit.module.only` behaves strange when using the module syntax that assigns tests to modules using the order of invocation, e.g.:

```javascript
QUnit.module.only('module 1');

QUnit.test('test in module 1', assert => {
  assert.ok(true, 'a successful assertion in module 1');
});

QUnit.module('module 2');

QUnit.test('test in module 2', assert => {
  assert.ok(true, 'a successful assertion in module 2');
});
```

Without `only` `test in module 1` belongs to `module 1` and `test in module 2` belongs to `module 2`. Using `only` in `module 1` the second test also seems to belong to `module 1` and is run (applying the wrong hooks if there would be any).

You can check this in this plunk: https://embed.plnkr.co/mh4VgJ0O6Dhp4H5APd3S/

**Comment by trentmwillis** [comment]
Thanks for the report + reproduction! Definitely a bug and likely just a combination we didn't test (I think all the tests use the module callback form).

**Comment by Arkni** [comment]
Hi,

I did some analysis and what I came with is the following: we can flag the current module (which is the one we attach `only` to) as `focused` by adding a new property to it. This property will be set to `true` as soon as the first ignored `module` is executed, and will then be used in conjuction with `focused` (inside `test.js`) to ignore tests of other modules.

In summary, here is the diff:

<details>
<summary>`src/core.js`</summary>

```diff
diff --git a/src/core.js b/src/core.js
index b2e9472..95230e5 100644
--- a/src/core.js
+++ b/src/core.js
@@ -115,6 +115,10 @@ function processModule( name, options, executeNow, modifiers = {} ) {
 // TODO: extract this to a new file alongside its related functions
 function module( name, options, executeNow ) {
 	if ( focused ) {
+		if ( !config.currentModule.focused ) {
+			config.currentModule.focused = true;
+		}
+
 		return;
 	}
 
@@ -130,6 +134,10 @@ function module( name, options, executeNow ) {
 
 module.only = function() {
 	if ( focused ) {
+		if ( !config.currentModule.focused ) {
+			config.currentModule.focused = true;
+		}
+
 		return;
 	}
 
@@ -143,6 +151,10 @@ module.only = function() {
 
 module.skip = function( name, options, executeNow ) {
 	if ( focused ) {
+		if ( !config.currentModule.focused ) {
+			config.currentModule.focused = true;
+		}
+
 		return;
 	}
 
@@ -158,6 +170,10 @@ module.skip = function( name, options, executeNow ) {
 
 module.todo = function( name, options, executeNow ) {
 	if ( focused ) {
+		if ( !config.currentModule.focused ) {
+			config.currentModule.focused = true;
+		}
+
 		return;
 	}
```
</details>

<details>
<summary>`src/test.js`</summary>

```diff
diff --git a/src/test.js b/src/test.js
index 097fd39..e5ab69a 100644
--- a/src/test.js
+++ b/src/test.js
@@ -658,7 +658,7 @@ function checkPollution() {
 
 // Will be exposed as QUnit.test
 export function test( testName, callback ) {
-	if ( focused ) {
+	if ( focused || config.currentModule.focused ) {
 		return;
 	}
 
@@ -671,7 +671,7 @@ export function test( testName, callback ) {
 }
 
 export function todo( testName, callback ) {
-	if ( focused ) {
+	if ( focused || config.currentModule.focused ) {
 		return;
 	}
 
@@ -686,7 +686,7 @@ export function todo( testName, callback ) {
 
 // Will be exposed as QUnit.skip
 export function skip( testName ) {
-	if ( focused ) {
+	if ( focused || config.currentModule.focused ) {
 		return;
 	}
 
@@ -700,7 +700,7 @@ export function skip( testName ) {
 
 // Will be exposed as QUnit.only
 export function only( testName, callback ) {
-	if ( focused ) {
+	if ( focused || config.currentModule.focused ) {
 		return;
 	}
```
</details><br>
If this solution looks good, I'll be happy to open a PR.

**Comment by smcclure15** [comment]
Resolved with #1436?

**Comment by Arkni** [comment]
@smcclure15 

Just tested with latest version and it doesn't seem to be fixed. See https://plnkr.co/edit/lLBZUFu7opjXpTVq?preview
